chore: add contributting guide (#31)
meindica/meindica: CONTRIBUTTING.md

# Guia de contribuição

> Oi, antes de tudo gostariamos de agradecer seu interesse em contribuir para o MeIndica!,

Reviewer: ```suggestion
Oi, antes de tudo gostaríamos de agradecer pelo seu interesse em contribuir para o MeIndica!,
```

ficamos bem felizes com você aqui.

> Também é importante ressaltar que você não precisa dominar o React pra contribuir, nós

Reviewer: ```suggestion
Também é importante ressaltar que você não precisa dominar o React para contribuir, nós
```

estamos aqui para ajudar e aprender juntos, então se você quer contribuir seja com documentação,

> ideias, sugestões reportar algum bug, sinta-se bem vindo.

Reviewer: ```suggestion
ideias, sugestões ou reportar algum bug, sinta-se bem vindo.
```

## Conteúdo

- Stack.
- Como iniciar o projeto.
- Abrindo um pull request.

### Stack

> A stack desse projeto é toda construida com o ecossistema do React, como framework estamos
> usando o [Gatsby](https://gatsbyjs.com) e para os componentes e estilos básicos estamos
> usando o [Chakra-ui](https://chakra-ui.com) e é basicamente isso.
> 
> Nossa base de dados é construida através de uma planilha do [google spreadsheets](https://www.google.com/sheets/about/).
> 
> Nossa hospedagem fica no [Netlify](https://www.netlify.com/) e á automáticamente deployada
> através de uma integração com o [Github](https://github.com).
> 

Reviewer: ```suggestion
A stack desse projeto é toda construída com o ecossistema do React, como framework estamos
usando o [Gatsby](https://gatsbyjs.com) e para os componentes e estilos básicos estamos
usando o [Chakra-ui](https://chakra-ui.com) e é basicamente isso.

Nossa base de dados é construída através de uma planilha do [Google Sheets](https://www.google.com/sheets/about/).

Nossa hospedagem fica no [Netlify](https://www.netlify.com/) e automaticamente _deployada_
através de uma integração com o [Github](https://github.com).

```

### Como iniciar o projeto

> Bem tendo conhecimento da nossa stack vamos ao passo a passo de como iniciar o projeto em si,
> o primeiro passo é você realizar um fork deste repositório, feito isso você pode clonar ser 
> fork no seu computador e a partir branch `main` criar uma nova branch:
> 

Reviewer: ```suggestion
Tendo conhecimento da nossa stack, vamos ao passo a passo de como iniciar o projeto em si,
o primeiro passo é você realizar um _fork_ deste repositório, após isso você pode clonar este 
fork no seu computador e criar uma nova branch a partir da branch `main`:

```

```bash
git clone git@github.com:<seu usuário>/meindica.git # aqui estamos assumindo que você já configurou o ssh para acessar o github.

> cd meindica

Reviewer: ```suggestion
cd meindica/
```

git checkout -b add-sua-funcionalidade
```

> A partir daqui você vai iniciar a configuração pra conseguir consumir os dados de uma planilha,
> então o primeiro passo é criar uma planilha no [google spreadsheets](https://www.google.com/sheets/about/)
> o formato dessa planilha precisa ser **exatamente igual** ao que usamos no MeIndica!
> incluindo o nome da aba, pra facilitar você pode [baixar este mock](docs/mock.xlsx).
> 
> **verifica se o nome da aba na tabela deve estar como: Respostas ao formulário 1**
> 
> Agora vamos a parte de configurar as chaves de API do Google.

Reviewer: ```suggestion
A partir daqui você vai iniciar a configuração para que consiga consumir os dados de uma planilha,
então o primeiro passo é criar uma planilha no [Google Sheets](https://www.google.com/sheets/about/),
o formato desta planilha precisa ser **exatamente igual** ao que usamos no MeIndica!
incluindo o nome da aba, pra facilitar você pode [baixar este mock](docs/mock.xlsx).

**verifique se o nome da aba na tabela está como: Respostas ao formulário 1**

Agora vamos a parte de configurar as chaves de API do Google.
```

> 1. Vá até o [congole do Google Apis](https://console.developers.google.com/) e clique em `Criar Projeto`.

Reviewer: ```suggestion
1. Vá até o [console do Google Apis](https://console.developers.google.com/) e clique em `Criar Projeto`.
```

2. Informe o nome do projeto e crie.

> 3. Com o projeto selecionado vá até credenciais e clieque em `Criar sentenciais` e selecione `Conta de serviço`.

Reviewer: ```suggestion
3. Com o projeto selecionado vá até credenciais e clique em `Criar credenciais` e selecione `Conta de serviço`.
```

4. Clique em `adicionar chave` selecione json e baixe o arquivo.

> 5. Na sua planilhe vá em `compartilhar` e adicione o client_email informado no arquivo json que você baixou.

Reviewer: ```suggestion
5. Na sua planilhe vá em `compartilhar` e adicione o `client_email` informado no arquivo json que você baixou.
```

6. Agora no projeto, copie o .env.example para um arquivo .env você pode usar `cp .env.example .env`
7. Adicione as configurações: `private_key`, `client_email` neste arquivo.

> 8. O valor de `SPREADSHEET_ID` pode ser encontrado na url da sua planilha.

Reviewer: ```suggestion
8. O valor de `SPREADSHEET_ID` pode ser encontrado na url da sua planilha. Exemplo: `https://docs.google.com/spreadsheets/d/<id-da-sua-planilha>/edit#gid=0`
```

9. O valor de WORKSHEET você deve colocar o nome que está como titulo da aba da sua tabela.

> Após feito isso você pode iniciar o projeto com: `yarn develop` se tudo correr como esperado
> você poderá acessar a aplicação em http://localhost:8000

Reviewer: ```suggestion
Após feito isso você pode iniciar o projeto com: `yarn develop` se tudo correr como esperado você poderá acessar a aplicação em [http://localhost:8000](http://localhost:8000)
```

### Abrindo um pull-request

Ao abrir o pull-request informe o que está sendo feito e se possível faça uma referência para a issue.

> Enfim eu acho que é isso.

Reviewer: O `README.md` está em inglês, seria melhor ter uma versão em português do README ou uma versão em inglês do `CONTRIBUTING.md` para não ficar misturado

Reviewer: Btw, o nome do arquivo está com dois T's

Reviewer: A propósito, ficou muito bom o guia :)

Author: A ideia é adicionar um guia em inglês e um readme em pt-br :grimacing: (exceto pela parte dos contribuidores porque é gerado automáticamente).

> Enfim eu acho que é isso.

Reviewer: ```suggestion
Teremos o prazer de ler o seu código e agradecemos pela sua contribuição! :)
```
